atef/config.py

```diff
@@ -144,18 +175,111 @@ class PreparedComparison:
+        be instantiated and a global PV-to-Signal cache will be utilized.
+
+        Parameters
+        ----------
+        config : PVConfiguration or DeviceConfiguration
+            The configuration.
+
+        client : happi.Client
+            A happi Client instance.
+
+        cache : dict of str to type[Signal]
+            The PV to signal cache.
+
+        Yields
+        ------
+        item : PreparedComparisonException or PreparedComparison
+            If an error occurs during preparation, a
+            PreparedComparisonException will be yielded in place of the
+            PreparedComparison.
+        """
+        if isinstance(config, PVConfiguration):
+            yield from PreparedSignalComparison._from_pv_config(config, cache=cache)
+        elif isinstance(config, DeviceConfiguration):
+            if client is None:
+                client = happi.Client.from_config()
+            for dev_name in config.devices:
+                try:
+                    device = util.get_happi_device_by_name(dev_name, client=client)
+                except Exception as ex:
+                    yield PreparedComparisonException(
+                        exception=ex,
+                        comparison=None,  # TODO
+                        name=config.name or config.description,
+                        identifier=dev_name,
+                        path=[
+                            config,
+                            dev_name,
+                        ],
+                    )
+                else:
+                    yield from PreparedSignalComparison._from_device_config(
+                        config=config,
+                        device=device,
+                    )
+        elif isinstance(config, ToolConfiguration):
+            yield from PreparedToolComparison._from_tool_config(config)
+        else:
+            raise NotImplementedError(f"Configuration type unsupported: {type(config)}")
+
+
+@dataclass
+class PreparedSignalComparison(PreparedComparison):
+    """
+    A unified representation of comparisons for device signals and standalone PVs.
+    """
+    #: The device the comparison applies to, if applicable.
+    device: Optional[ophyd.Device] = None
+    #: The signal the comparison is to be run on.
+    signal: Optional[ophyd.Signal] = None
+    #: The signal the comparison is to be run on.
+    data: Optional[Any] = None
+
+    async def get_data(self) -> Any:
+        """
+        Get the provided signal's data according to the comparison's reduction
+        configuration.
+
+        Returns
+        -------
+        data : Any
+            The acquired data.
+
+        Raises
+        ------
+        TimeoutError
+            If unable to connect or retrieve data from the signal.
+        """
+        if self.signal is None:
+            raise ValueError("Signal instance unset")
+        # TODO: async?
+        self.data = self.comparison.get_data_for_signal(self.signal)
```

Comment:
I wonder if it's possible or desirable to build some async paths into the ophyd `Signal` classes for these sorts of applications.
```
class Signal(...):
    async def get_async(self, **kwargs):
        return some_future_suspiciously_similar_to_a_status
```
(if it's not super clear I haven't worked much with asyncio directly)

Reply:
There might be something we can do here, but it'll take a bit of investigation.

1. Things that can happen asynchronously - and report asynchronous status via callbacks can be relatively easily turned asyncio-compatible with some cross-communication between a callback and an asyncio Task
2. Things that have busy loops, such as the [poor mechanism pyepics uses](https://github.com/pyepics/pyepics/blob/762b96fe10dfb0194e3a5ae3eb9df8fe0a04b2c1/epics/ca.py#L1523-L1529) in the background (and ophyd inherits necessarily) 

For the pyepics example in (2), the hack would have to happen at roughly the `epics.ca` or `epics.PV` level, making it not so compelling/desirable to modify